nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_dataset.py

```diff
@@ -38,6 +38,7 @@ def __init__(
         seed: int = 1234,
         context_key: str = "text",
         label_key: str = "answer",
+        reference_key: str = None,
```

Comment:
why do you need this extra field? why not use label_key are the reference for inferencing?

Reply:
Currently, our dataset json file only has column for `context_key` and `label_key` both are in string format. However, users may want to pass other columns to the prediction files, e,g. IDs, other labels, etc. In this scenario, I use `reference_key` to pass the column with multiple labels (some tasks has multiple reference answers than just one). I am not sure do we have another good way to pass columns to prediction files since we may face order issue due to using distributed sampler.

Reply:
do you mean this is just for including some metadata for each test input? i.e. the model does not use this information in any way?

Reply:
for multiple possible correct labels for evaluation a better way is to allow a list of labels instead of a single string.

Reply:
I currently use `reference_key` to pass multiple answers and calculate metrics between predictions and answers. 
There are two issues related to these changes.
1. We want to pass some metadata for each test input.
2. We want to calculate metrics with multiple answers.
For 1., I am not sure what is the best way for the implementation. 
For 2., I can change to directly use `label_key` as the answer and calculate metrics, which may make more sense. Then user can calculate their metrics if we have implementation 1.

Reply:
I remove this field and simply copy all columns except context_key and label_key to the prediction file.

Reply:
Whats the difference between the old reference_key and the new query_key that you've added?

Reply:
`reference_key` is used to pass some key-value metadata from dataset jsonl file to prediction jsonl file. I remove this by just passing all key-value pairs except context_key and label_key.

`query_key` is used to have the flexibility to construct prompts. Previously, we only have `context_key` and the query part (questions) is at the front or the end of the context. It is hard to truncate the contexts. Therefore, we add one key for query, so user can construct the prompt based on their preference. 

Here is an example from Qasper in Scrolls:
context_key: `Introduction\\nCategories such as animal or furniture are fundamental cognitive building blocks allowing humans to efficiently represent ...`
query_key: `do language share categories?`
label_key: `Yes`
prompt: `{context} <extra_id_1>{query}<extra_id_2>\n{label}`